tests/karma.js

```diff
@@ -2151,6 +2155,1028 @@ beforeAll(function (done) {
+              '  <item_type>file</item_type>\n' +
+              '  <file_target>' + file[i] + '</file_target>\n' +
+              '  <share_with>test123</share_with>\n' +
+              '  <share_with_displayname>test123</share_with_displayname>\n' +
+              '  <share_with_additional_info/>\n' +
+              '  <attributes/>\n' +
+              ' </data>\n' +
+              '</ocs>'
+          }
+        })
+      }
+    })
+    .then(() =>
+      provider.addInteraction({
+        uponReceiving: 'a link share POST request with link name',
+        withRequest: {
+          method: 'POST',
+          path: Pact.Matchers.term({
+            matcher: '.*\\/ocs\\/v1\\.php\\/apps\\/files_sharing\\/api\\/v1\\/shares$',
+            generate: '/ocs/v1.php/apps/files_sharing/api/v1/shares'
+          }),
+          headers: {
+            authorization: 'Basic YWRtaW46YWRtaW4=',
+            Origin: origin
+          },
+          body: 'shareType=3' + '&path=%2F' + config.testFolder + '&name=%C3%96ffentlicher+Link'
+        },
+        willRespondWith: {
+          status: 200,
+          headers: {
+            'Access-Control-Allow-Origin': '*',
+            'Access-Control-Allow-Headers': accessControlAllowHeaders,
+            'Access-Control-Allow-Methods': 'GET,OPTIONS,POST,PUT,DELETE,MKCOL,PROPFIND,PATCH,PROPPATCH,REPORT'
+          },
+          body: '<?xml version="1.0"?>\n' +
+            '<ocs>\n' +
+            ' <meta>\n' +
+            '  <status>ok</status>\n' +
+            '  <statuscode>100</statuscode>\n' +
+            '  <message/>\n' +
+            '  <totalitems></totalitems>\n' +
+            '  <itemsperpage></itemsperpage>\n' +
+            ' </meta>\n' +
+            ' <data>\n' +
+            '  <id>9</id>\n' +
+            '  <share_type>3</share_type>\n' +
+            '  <uid_owner>admin</uid_owner>\n' +
+            '  <displayname_owner>admin</displayname_owner>\n' +
+            '  <permissions>1</permissions>\n' +
+            '  <parent/>\n' +
+            '  <expiration/>\n' +
+            '  <uid_file_owner>admin</uid_file_owner>\n' +
+            '  <displayname_file_owner>admin</displayname_file_owner>\n' +
+            '  <additional_info_owner/>\n' +
+            '  <additional_info_file_owner/>\n' +
+            '  <path>/testFolder</path>\n' +
+            '  <item_type>folder</item_type>\n' +
+            '  <file_target>/testFolder</file_target>\n' +
+            '  <name>Öffentlicher Link</name>\n' +
+            '  <url>http://localhost/oc/s/H1wiMY2oDYmnWdC</url>\n' +
+            '  <attributes/>\n' +
+            ' </data>\n' +
+            '</ocs>'
+        }
+      }))
+    .then(() => {
+      provider.addInteraction({
+        uponReceiving: 'a request to update the permissions of a public link share',
+        withRequest: {
+          method: 'PUT',
+          path: Pact.Matchers.regex({
+            matcher: '.*\\/ocs\\/v1\\.php\\/apps\\/files_sharing\\/api\\/v1\\/shares\\/9$',
```

Comment:
make it more general e.g.
```suggestion
            matcher: '.*\\/ocs\\/v1\\.php\\/apps\\/files_sharing\\/api\\/v1\\/shares\\/\d+$',
```

Reply:
fixed